lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -136,6 +128,54 @@ def build_client(options)
       Elasticsearch::Client.new(client_options)
     end
 
+    HOSTNAME_PORT_REGEX=/\A(?<hostname>([A-Za-z0-9\.\-]+))(:(?<port>\d+))?\Z/
+    # Parse a configuration host to a normalized URL
+    def host_to_url(host, ssl=nil, path=nil)
```

Comment:
For this specific case, I don't have strong feelings, so as a way to not add noise to a review, I retract my comment :)